Gtest/sensors_fixture.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <thread>
+#include <chrono>
+#include "sensors_fixture.h"
+
+#define SUPPORTED_SENSORS_NUMBER 9
+#define INVALID_SENSOR_TYPE     -1
+#define INVALID_SENSOR_HANDLE   -1
+#define ENABLED                  1
+#define DISABLED                 0
+#define UNUSED_FLAG             -1
+#define SAMPLE_PERIOD_NS         50*1000000
+#define AT_LEAST_ONE_EVENT       1
+#define SUCCESS                  0
+#define FAILED                   -EINVAL
+
+#define LOG_TAG "SensorsFixture"
+
+extern int sensor_poll_events(struct sensors_poll_device_t* dev0, sensors_event_t* data, int count);
+extern int sensor_activate(struct sensors_poll_device_t* dev0, int handle, int enabled);
+extern int sensor_batch(struct sensors_poll_device_1* dev0, int handle, int flags __unused, \
+                        int64_t sampling_period_ns, int64_t max_report_latency_ns __unused);
+extern int sensor_set_delay(struct sensors_poll_device_t* dev0, int handle __unused, int64_t ns);
+extern int sensor_flush(struct sensors_poll_device_1* dev0, int handle);
+extern int sensor_close(struct hw_device_t* dev0);
+
+SensorsFixture::SensorsFixture() {
+    m_test_dev = new SensorDevice();
+}
+
+SensorsFixture::~SensorsFixture() {
+    if(m_test_dev) {
+        delete m_test_dev;
+        m_test_dev = nullptr;
+    }
+}
+
+TEST_F(SensorsFixture, SocketConnectionCheck)
+{
+    this_thread::sleep_for(std::chrono::milliseconds(2000)); //Make sure client connected to server
+    ASSERT_TRUE(m_sensors_client.is_connected());
+}
+
+TEST_F(SensorsFixture, SupportedSensorsNumberCheck)
+{
+    ASSERT_EQ(SUPPORTED_SENSORS_NUMBER, m_sensors_helper.get_supported_sensors_num()); // Server Check
+}
+
+TEST_F(SensorsFixture, SensorsTypeCheck)
+{
+    ASSERT_FALSE(m_sensors_helper.is_supported_type(INVALID_SENSOR_TYPE));
```

Comment:
Target of this test is to query supported sensors from sensor_vhal.cpp. 
For this we can use m_test_dev->get_handle_from_type(). If it returns -1 then its an unsupported sensor